src/app-layout/visual-refresh/trigger-button.tsx

```diff
@@ -1,24 +1,61 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
-import React from 'react';
+import React, { Ref, useEffect, useState } from 'react';
 import clsx from 'clsx';
 
 import { ButtonProps } from '../../button/interfaces';
+import { InternalButton } from '../../button/internal';
 import { IconProps } from '../../icon/interfaces';
 import Icon from '../../icon/internal';
+import Tooltip from '../../internal/components/tooltip';
 
 import styles from './styles.css.js';
 
 export interface TriggerButtonProps {
   ariaLabel?: string;
   className?: string;
+  /**
+   * The name of the Icon.
+   *
+   * An icon is neccessary for this component to render. If the name is not given there must be a iconSvg prop passed or the component will not render
+   */
   iconName?: IconProps.Name;
+  /**
+   * A svg node for a custom icon.
+   *
+   * An icon is neccessary for this component to render. If the SVG is not given there must be a iconName prop passed or the component will not render
+   */
   iconSvg?: React.ReactNode;
   ariaExpanded: boolean | undefined;
+  /**
+   * Set to true if this component used in a mobile layout
+   *
+   * It will then render a different InternalButton that is optimumn for the mobile experience
+   */
+  isMobile?: boolean;
   ariaControls?: string;
+  disabled?: boolean;
+  /**
+   * If the button is expected to have a tooltip. When false it will not set the event listeners
+   *
+   * defaults to false
+   */
+  hasTooltip?: boolean;
```

Comment:
Added this prop so trigger-buttons that do not have tooltips do not create event listeners